src/components/root.js

```diff
@@ -27,13 +28,38 @@ class Root extends Component {
     this.ref = React.createRef()
   }
   componentDidMount () {
-    this.props.match.params.id
+    this.props.match.params.uuid &&
+    this.props.match.params.id &&
+    this.props.match.params.type_shared &&
+    this.props.match.params.type_access
```

Comment:
it is looking very hectic.
store params in a variable to shorten "this.props.match.params" to only "params"